src/Hamiltonians/geometry.jl

```diff
@@ -222,21 +276,118 @@ Base.size(off::Displacements) = (length(off.geometry),)
     end
 end
 
-"""
-    neighbor_site(geom::CubicGrid, site, i)
-
-Find the `i`-th neighbor of `site` in the geometry. If the move is illegal, return 0.
-
-See also [`CubicGrid`](@ref).
-"""
 function neighbor_site(g::CubicGrid{D}, mode, chosen) where {D}
     # TODO: reintroduce LadderBoundaries small dimensions
     return g[g[mode] + Directions(D)[chosen]]
 end
 
+num_neighbors(::CubicGrid{D}) where {D} = 2D
+
 function BitStringAddresses.onr(address, geom::CubicGrid{<:Any,S}) where {S}
     return SArray{Tuple{S...}}(onr(address))
 end
 function BitStringAddresses.onr(address::CompositeFS, geom::CubicGrid)
     return map(fs -> onr(fs, geom), address.components)
 end
+
+"""
+    HoneycombLattice((height, width), fold=(true, true))
+
+A honeycomb lattice where each site has three neighbors. If periodic, each dimension of the
+lattice must be divisible by 2.
+
+A 4×4 `HoneycombLattice` is indexed as follows.
+```
+  ╲    ╱    ╲    ╱
+   1──5      9─13
+  ╱    ╲    ╱    ╲
+─2      6─10     14─
+  ╲    ╱    ╲    ╱
+   3──7     11─15
+  ╱    ╲    ╱    ╲
+─4      8─12     16─
+  ╲    ╱    ╲    ╱
+```
+"""
+struct HoneycombLattice{Dims,Fold} <: Geometry{2}
+    function HoneycombLattice(dims::Tuple{Int,Int}, fold::Tuple{Bool,Bool}=(true, true))
+        if any(<(2), dims)
+            throw(ArgumentError("All dimensions must be at least 2 in size"))
+        end
+        if fold[1] && isodd(dims[1]) || fold[2] && isodd(dims[2])
+            throw(ArgumentError("Periodic dimensions must be even in size"))
+        end
+        return new{dims,fold}()
+    end
+end
+HoneycombLattice(h, w, fold::Tuple{Bool,Bool}=(true, true)) = HoneycombLattice((h, w), fold)
+
+function Base.show(io::IO, geom::HoneycombLattice)
+    h, w = size(geom)
+    fold = periodic_dimensions(geom)
+    print(io, "HoneycombLattice($h, $w, $fold)")
+end
+
+Base.size(::HoneycombLattice{Dims}) where {Dims} = Dims
+periodic_dimensions(::HoneycombLattice{<:Any,Fold}) where {Fold} = Fold
+
+function neighbor_site(geom::HoneycombLattice, mode, chosen)
+    i, j = geom[mode]
+    if chosen ≤ 2
+        target = SVector(i + ifelse(chosen == 1, 1, -1), j)
+    else
+        target = SVector(i, j + ifelse(isodd(i + j), -1, 1))
+    end
+    source = geom[mode]
+    return geom[target]
+end
+
+num_neighbors(::HoneycombLattice) = 3
+
+"""
+    HexagonalLattice((height, width), fold=(true, true))
```

Comment:
```suggestion
    HexagonalLattice((height, width), fold=(true, true)) <: Geometry{2}
```
Again, please link back to `Geometry`.

Reply:
```suggestion
    HexagonalLattice(height, width, fold=(true, true)) <: Geometry{2}
```
Should we drop the parenthesis to make it consistent with the `show` method?